Lib/test/test_type_params.py

```diff
@@ -823,6 +823,39 @@ def meth[__U](self, arg: __T, arg2: __U):
 
         self.assertEqual(Foo.Alias.__value__, (T, V))
 
+    def test_no_leaky_mangling_in_module(self):
+        ns = run_code("""
+            __before = "before"
+            class X[T]: pass
+            __after = "after"
+        """)
+        self.assertEqual(ns["__before"], "before")
+        self.assertEqual(ns["__after"], "after")
+
+    def test_no_leaky_mangling_in_function(self):
+        ns = run_code("""
+            def f():
+                class X[T]: pass
+                _X_foo = 2
+                __foo = 1
+                assert locals()['__foo'] == 1
+                return __foo
+        """)
+        self.assertEqual(ns["f"](), 1)
+
+    def test_no_leaky_mangling_in_class(self):
+        ns = run_code("""
+            class Outer:
+                __before = "before"
+                class Inner[T]:
+                    __x = "inner"
```

Comment:
Is it a good idea to test `__before = "class"; __after = "class"` names in the class scope here? Or do we have them tested?

Reply:
You mean in the inner class? That wouldn't really be relevant here since there is nothing else interesting going on in the inner class body.